updating/updating_a_cluster/update-using-custom-machine-config-pools.adoc

```diff
@@ -12,21 +12,42 @@ WARNING: This assembly has been moved into a subdirectory for 4.14+. Changes to
 
-During your first maintenance window, you would pause the MCP for *workerpool-A*, *workerpool-B*, and *workerpool-C*, then initiate the cluster update. This updates components that run on top of {product-title} and the 10 nodes which are members of the *workerpool-canary* MCP, because that pool was not paused. The other three MCPs are not updated, because they were paused.  If for some reason, you determine that your cluster or workload health was negatively affected by the *workerpool-canary* update, you would then cordon and drain all nodes in that pool while still maintaining sufficient capacity until you have diagnosed the problem. When everything is working as expected, you would then evaluate the cluster and workload health before deciding to unpause, and thus update, *workerpool-A*, *workerpool-B*, and *workerpool-C* in succession during each additional maintenance window.
+[id="example_{context}"]
+== Example Canary update strategy
 
-While managing worker node updates using custom MCPs provides flexibility, it can be a time-consuming process that requires you execute multiple commands. This complexity can result in errors that can affect the entire cluster. It is recommended that you carefully consider your organizational needs and carefully plan the implemention of the process before you start.
+The following example describes a canary update strategy where you have a cluster with 100 nodes with 10% excess capacity, you have maintenance windows that must not exceed 4 hours, and you know that it takes no longer than 8 minutes to drain and reboot a worker node.
 
-[NOTE]
-====
-It is not recommended to update the MCPs to different {product-title} versions. For example, do not update one MCP from 4.y.10 to 4.y.11 and another to 4.y.12.
-This scenario has not been tested and might result in an undefined cluster state.
-====
+.Defining custom machine config pools
+
+In order to organize the worker node updates into separate stages, you could begin by defining the following MCPs:
+
+* *workerpool-canary* with 10 nodes
+* *workerpool-A* with 30 nodes
+* *workerpool-B* with 30 nodes
+* *workerpool-C* with 30 nodes
+
+.Updating the canary worker pool
+
+During your first maintenance window, you would pause the MCPs for *workerpool-A*, *workerpool-B*, and *workerpool-C*, and then initiate the cluster update.
+This updates components that run on top of {product-title} and the 10 nodes that are part of the *workerpool-canary* MCP, because that pool was not paused.
+The other three MCPs are not updated, because they were paused.
+
+.Determining whether to proceed with the remaining worker pool updates
+
+If for some reason you determine that your cluster or workload health was negatively affected by the *workerpool-canary* update, you would then cordon and drain all nodes in that pool while still maintaining sufficient capacity until you have diagnosed the problem.
```

Comment:
I'm trying to understand the statement. If there's a negative impact on the cluster or workload health due to the update of the workerpool-canary, we are suggesting users to cordon and drain all nodes in the workerpool-canary so that the workload will be migrated to nodes in other work pools. Right? 

@jianlinliu @rioliu-rh PTAL. Thank you.

Reply:
I think the issue here is, when the update is stuck at a node (`maxUnavailable = 1`), the pool is degraded, we need to do RCA to figure out the root cause and apply a fix on it, then rest of nodes will be updated (cordon, drain, update, reboot etc) automatically by MCO

Reply:
The documentation is suggesting that if the user find out that the new update is not working for the application after updating the first machine config pool, the user can cordon and drain the application from nodes with new version. Because the cluster has sufficient capacity the quality of service wont be impacted. Also it is assumed that update is successful but the application might not be compatible with the worker nodes with the new version because of some API incompatibility etc.

Reply:
Is there any doc change that should be incorporated here? Perhaps making it more explicit that we expect users to resolve the issue before resuming their update? Maybe the following:
```suggestion
If for some reason you determine that your cluster or workload health was negatively affected by the *workerpool-canary* update, you would then cordon and drain all nodes in that pool while still maintaining sufficient capacity until you have diagnosed and resolved the problem.
Once the issue is resolved, you would then uncordon the nodes to have workloads scheduled to them again.
```

Reply:
The update^ looks good to me.